src/data-model-providers/endpoint/tests/TestEndpointProviderRegistry.cpp

```diff
@@ -0,0 +1,319 @@
+/*
+ *    Copyright (c) 2025 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#include <pw_unit_test/framework.h>
+
+#include <app/data-model-provider/MetadataTypes.h>
+#include <data-model-providers/endpoint/EndpointProviderRegistry.h>
+#include <data-model-providers/endpoint/SpanEndpointProvider.h>
+
+#include <algorithm> // For std::shuffle
+#include <list>      // For std::list in stress test
+#include <random>    // For std::default_random_engine
+#include <vector>    // For std::vector in stress test
+using namespace chip;
+using namespace chip::app;
+
+TEST(TestEndpointProviderRegistry, CreateAndDestroy)
+{
+    EndpointProviderRegistry registry;
+
+    // Create a provider
+    auto build_pair_provider =
+        SpanEndpointProvider::Builder(1).SetComposition(DataModel::EndpointCompositionPattern::kFullFamily).build();
```

Comment:
nit: have 1 be some kTestEndpointId constant instead, so we have less hardcoding. This will make it clear that we want the same ID below for get and unregister.